Assignment one

nohagarem opened the pull request:
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
Writing queries to be able to analyze data

## What did you learn from the changes you have made?
I learned how to select columns from tables satisfying certain conditions, how to join tables, aggregate data, create temp tables and CTEs.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
For the last task on the sql assignment, I was thinking of using WHERE and LIKE with the wild card % to find the appropriate market_dates. I actually did that first, but then noticed that we were being asked to use the preceding query as a base for the last task.

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
I had a bit of trouble with the very last task in the sql assignment. I started by using strftime in the WHERE clause and wasn't getting any results. I realized that the result of strftime is not yet available to the WHERE statement and ended up using a CTE so I can use the preceding query as a base.

## How were these changes tested?
I ran each query separately and took a look at the results. I compared that with my predictions after browsing the data in the appropriate tables.

## A reference to a related issue in your repository (if applicable)

## Checklist
- [x ] I can confirm that my changes are working as intended

nohagarem committed:
Added png inside assignment1 file for Task1

nohagarem committed:
added some of my thoughts for the ethics task

nohagarem committed:
Took my thoughts for task 4 out of the copy box

nohagarem committed:
Added responses to the assignment1.sql file

kelichiu commented on 02_activities/assignments/assignment1.sql:
```diff
@@ -27,30 +35,52 @@ filtered by vendor IDs between 8 and 10 (inclusive) using either:
 */
 -- option 1
-
+SELECT product_id, vendor_id, market_date, customer_id, quantity, cost_to_customer_per_qty, transaction_time, quantity*cost_to_customer_per_qty  AS price,
```
Remove the last `,`

kelichiu commented on 02_activities/assignments/assignment1.sql:
```diff
@@ -27,30 +35,52 @@ filtered by vendor IDs between 8 and 10 (inclusive) using either:
 
 -- option 2
-
+SELECT product_id, vendor_id, market_date, customer_id, quantity, cost_to_customer_per_qty, transaction_time quantity*cost_to_customer_per_qty  AS price,
```
Missing `,` between columns, and extra `,` for the last column

kelichiu commented on 02_activities/assignments/assignment1.sql:
```diff
@@ -97,4 +155,17 @@ Remember that money spent is quantity*cost_to_customer_per_qty.
 HINTS: you will need to AGGREGATE, GROUP BY, and filter...
 but remember, STRFTIME returns a STRING for your WHERE statement!! */
-
+WITH my_purchase AS(
```
Great use of CTE here!

kelichiu reviewed (changes requested):
## Point: 29/30
## Section 1: ✅ Complete
## Section 2: ☑️ Partial
### Where Question 2
#### Todo:

- Option 1, there is an extra `,`, please remove it
- Option 2, there is an extra `,` and a missing ',' between columns, please revisit

## Section 3: ✅ Complete
Great use of CTE!
## Section 4: ✅ Complete

nohagarem committed:
Update assignment1.sql

For the second WHERE query, removed an extra , from option 1, and removed an extra , and added a missing , in option 2

kelichiu reviewed (approved):
## 2nd Assessment Point: 30/30
## Section 1: ✅ Complete
## Section 2: ✅ Complete
## Section 3: ✅ Complete
## Section 4: ✅ Complete
